src/backend/tests/stats/test_stats_save_result.py

```diff
@@ -0,0 +1,53 @@
+from flask import g, request
+
+
+import expungeservice
+from expungeservice.database.db_util import rollback_errors
+from expungeservice import stats
+from tests.factories.crawler_factory import CrawlerFactory
+from expungeservice.database import get_database
+from tests.endpoints.endpoint_util import EndpointShared
+from flask.views import MethodView
+
+
+class TestStatsView(MethodView):
+    """
+    To get the direct call to stats.save_result() to work,
+    this endpoint wrapper is needed so that the flask.current_user
+    object gets populated with user_id
+    """
```

Comment:
Can you elaborate on how `flask.current_user` gets populated with this mock endpoint and why it doesn't without it?

Reply:
Thanks for asking! It did seem weird to require this workaround, so I looked around a bit more and found a better solution. 

Before I decided to wrap the function call in an endpoint, I was trying to use the usual testing wrapper code for flask:

```
with flaskr.app.test_client() as client:
        with flaskr.app.app_context():
                test_stuff()
````

To the best of my understanding: this would create a flask app context that was distinct from the app context created from the test's `login()` call, and the `current_user` object isn't carried over between contexts. flask_login normally handles the `current_user` data, but that underlying functionality seemed to only kick in when making an actual endpoint call on the app, and not from just using the `with` context handler. 

Makes enough sense, I guess.

I found that I